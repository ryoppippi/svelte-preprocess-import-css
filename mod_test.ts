import { join } from "node:path";
import { assertSnapshot } from "@std/testing/snapshot";
import { preprocess } from "svelte/compiler";
import { importCSSPreprocess } from "./mod.ts";

async function load(_filename: string) {
  const filename = join(
    import.meta.dirname as string,
    _filename,
  );

  const source = Deno.readTextFileSync(filename);
  const { code } = await preprocess(
    source,
    [importCSSPreprocess()],
    { filename },
  );

  return code;
}

Deno.test(
  "example svelte main",
  async function (t) {
    const code = await load("./test_project/Main.svelte");
    await assertSnapshot(t, code);
  },
);

Deno.test(
  "example svelte partial",
  async function (t) {
    const code = await load("./test_project/Partial.svelte");
    await assertSnapshot(t, code);
  },
);

Deno.test(
  "example svelte rename",
  async function (t) {
    const code = await load("./test_project/Rename.svelte");
    await assertSnapshot(t, code);
  },
);
