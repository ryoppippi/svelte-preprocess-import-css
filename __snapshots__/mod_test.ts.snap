export const snapshot = {};

snapshot[`example svelte main 1`] = `
\`<script>
  let name = 'world';
</script>

<div id="message"> hello {name} </div>

<style>
  
div {
  color: red;
  background: blue;
}



  #message {
    text-decoration: underline;
  }
</style>
\`
`;

snapshot[`example svelte partial 1`] = `
\`<script>
  let name = 'world';
</script>

<div class="m2">This is a paragraph</div>
<p id="message">This is a paragraph</p>

<style>
  
div {
  color: red;
  background: blue;
}


  
p {
  color: blue;
}

.m2 {
  background: yellow;
}



 #message {
    text-decoration: underline;
  }
</style>
\`
`;

snapshot[`example svelte rename 1`] = `
'<div class="m3">This is a paragraph</div>

<style>
  
.m3 {
  background: yellow;
}


</style>
'
`;
